libsplinter/src/network/auth/state_machine/challenge_v1.rs

```diff
@@ -0,0 +1,268 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use crate::network::auth::ManagedAuthorizationState;
+
+use super::{
+    AuthorizationActionError, AuthorizationLocalAction, AuthorizationLocalState,
+    AuthorizationRemoteAction, AuthorizationRemoteState, Identity,
+};
+
+#[derive(PartialEq, Debug, Clone)]
+pub(crate) enum ChallengeAuthorizationLocalState {
```

Comment:
The organization of this file makes it hard to read/understand -- primarily the ordering of the items within. Currently:

- enum ChallengeAuthorizationLocalState
- impl Display for ChallengeAuthorizationLocalState
- enum ChallengeAuthorizationLocalAction
- impl Display for ChallengeAuthorizationLocalAction
- enum ChallengeAuthorizationRemoteState
- impl Display for ChallengeAuthorizationRemoteState
- enum ChallengeAuthorizationRemoteAction
- impl Display for ChallengeAuthorizationRemoteAction
- impl ChallengeAuthorizationLocalState
- impl ChallengeAuthorizationRemoteState

The Display statements are essentially noise since they don't help understand the code, yet are given preference over the actual impls. The impl of the enum should be close to the enum itself.  Either:

- enum ChallengeAuthorizationLocalState
- impl ChallengeAuthorizationLocalState
- impl display for ChallengeAuthorizationLocalState
- enum ChallengeAuthorizationLocalAction
- impl Display for ChallengeAuthorizationLocalAction
- enum ChallengeAuthorizationRemoteState
- impl ChallengeAuthorizationRemoteState
- impl Display for ChallengeAuthorizationRemoteState
- enum ChallengeAuthorizationRemoteAction
- impl Display for ChallengeAuthorizationRemoteAction

or as separate modules (my personal preference):

- ChallengeAuthorizationLocalState in challenge_v1::local_state
- ChallengeAuthorizationLocalAction in challenge_v1::local_action
- ChallengeAuthorizationRemotelState in challenge_v1::remote_state
- ChallengeAuthorizationRemoteAction in challenge_v1::remote_action

or 

- all local stuff in challenge_v1::local
- all remote stuff in challenge_v1::remote

And then "pub(crate) use" statements to bring them into the challenge_v1 module.

Reply:
Updated for your preferred option above.